### PR TITLE
V2 : "Get the code" button fixed. Issue solettaproject/soletta-dev-app#62

### PR DESCRIPTION
**Pull request by kamidipreetham**
Changes since V1:
- Removed ExamplesView.js.
- Removed all ExamplesView.js references.
- Corrected indentation and spaces in code.
- Added error message when loading file from local source in Cheat Sheet.

[V1](https://github.com/solettaproject/soletta/pull/1835):
-Corrected copy code functionality using execCommand API.
-Moved all of ExamplesView.js code inline as execCommand API doesn't work in injected scripts.
Signed-off-by: Kamidi Preetham <kamidi@live.com>

### CODE REVIEW

**Comment by brunobottazzini** [comment]
@kamidipreetham, Your commit got a little bit messy. I think it is because of the squash.

Try rewriting and adding the signed off in the last line, where it used to be.

**Comment by brunobottazzini** [comment]
Also your patch is outdated with the master and has a conflicts to it.

Rebase it with the soletta master.

File: doc/node-types-html/index.html.in [hunk 1]
```diff
@@ -15,10 +15,225 @@
...
+            title.text(_data['name']);
+            description.text(_data['description']);
+            info.append(title);
+            info.append(description);
+            // Check for local load file requests
+            var x = location.protocol;
+            var y = /^file?:/i;
+            if(y){
+                alert("Error: Running cheat-sheet from local source! \nPlease run cheat-sheet from server");
+            }
```
Comment (brunobottazzini) [inline doc/node-types-html/index.html.in]: This error message shouldn't be here. Let's warn the user about this error when he clicks in the copy button

File: doc/node-types-html/index.html.in [hunk 2]
```diff
@@ -15,10 +15,225 @@
...
+            /* loading title and description */
+            var info = $('#example-info');
+            var title = $('<h1></h1>');
+            var description = $('<p></p>');
+            title.text(_data['name']);
+            description.text(_data['description']);
+            info.append(title);
+            info.append(description);
+            // Check for local load file requests
+            var x = location.protocol;
```
Comment (brunobottazzini) [inline doc/node-types-html/index.html.in]: Is this var used ?

File: doc/node-types-html/index.html.in [hunk 3]
```diff
@@ -15,10 +15,225 @@
...
+         * COPY EXAMPLE CODE
+         * Implement the algorithm to read the example
+         * code here. The var code already stores url 
+         * when creating the Entry in the main loop
+         * --------------------------------------------------------
+         */
+        function copyExampleCode(event){
+            var x = document.getElementById('example-info-code');
+            selecttext(x);
+            y=copycode();
```
Comment (brunobottazzini) [inline doc/node-types-html/index.html.in]: Split the returned var from the function. 

```y =  copycode();```

Also where do you declare this var ? Why not using it to verify if the copycode was successful ?

File: doc/node-types-html/index.html.in [hunk 4]
```diff
@@ -15,10 +15,225 @@
...
+            console.log(y);
```
Comment (brunobottazzini) [inline doc/node-types-html/index.html.in]: We don't need this console.log here

File: doc/node-types-html/index.html.in [hunk 5]
```diff
@@ -15,10 +15,225 @@
...
+            var getCodeButton = $('<img></img>');
+            getCodeButton.attr('src','images/button_get_code.png');
+            getCodeButton.attr('id','example-get-code');
+            info.append(getCodeButton);
+            var code = $('#example-image').data('code'); 
+            $.get(String(code),function(data){              //Getting code from path
+                hidden_code.html(String(data));
+            }); 
+            getCodeButton.on('click',copyExampleCode);
+            }
```
Comment (brunobottazzini) [inline doc/node-types-html/index.html.in]: Indent all this code inside the else

**Comment by kamidipreetham** [comment]
V3:
- Error message now pops up when user clicks "Get the code" button if user loads Cheat Sheet from local source.
- Removed unnecessary console.log()
- Added proper signed off message.

**Pull request closed without merging**